src/Storages/ObjectStorageQueue/ObjectStorageQueueUnorderedFileMetadata.cpp

```diff
@@ -37,56 +37,66 @@ ObjectStorageQueueUnorderedFileMetadata::ObjectStorageQueueUnorderedFileMetadata
-        PROCESSED_PATH_DOESNT_EXIST = 0,
-        /// node_name is not within failed persistent nodes
-        FAILED_PATH_DOESNT_EXIST = 2,
-        /// node_name ephemeral processing node was successfully created
-        CREATED_PROCESSING_PATH = 4,
-        /// update processing id
-        SET_PROCESSING_ID = 6,
-    };
-
     const auto zk_client = getZooKeeper();
     processing_id = node_metadata.processing_id = getRandomASCIIString(10);
     auto processor_info = getProcessorInfo(processing_id.value());
 
-    Coordination::Requests requests;
-    requests.push_back(zkutil::makeCreateRequest(processed_node_path, "", zkutil::CreateMode::Persistent));
-    requests.push_back(zkutil::makeRemoveRequest(processed_node_path, -1));
-    requests.push_back(zkutil::makeCreateRequest(failed_node_path, "", zkutil::CreateMode::Persistent));
-    requests.push_back(zkutil::makeRemoveRequest(failed_node_path, -1));
-    requests.push_back(zkutil::makeCreateRequest(processing_node_path, node_metadata.toString(), zkutil::CreateMode::Ephemeral));
+    while (true)
+    {
+        Coordination::Requests requests;
+        const size_t processed_path_doesnt_exist = 0;
+        zkutil::addCheckNotExistsRequest(requests, *zk_client, processed_node_path);
+        const size_t failed_path_doesnt_exist = requests.size();
+        zkutil::addCheckNotExistsRequest(requests, *zk_client, failed_node_path);
 
-    requests.push_back(
-        zkutil::makeCreateRequest(
-            processing_node_id_path, processor_info, zkutil::CreateMode::Persistent, /* ignore_if_exists */true));
-    requests.push_back(zkutil::makeSetRequest(processing_node_id_path, processor_info, -1));
 
-    Coordination::Responses responses;
-    const auto code = zk_client->tryMulti(requests, responses);
-    auto is_request_failed = [&](RequestType type) { return responses[type]->error != Coordination::Error::ZOK; };
+        const auto created_processing_path = requests.size();
+        requests.push_back(zkutil::makeCreateRequest(processing_node_path, node_metadata.toString(), zkutil::CreateMode::Ephemeral));
 
-    if (code == Coordination::Error::ZOK)
-    {
-        const auto * set_response = dynamic_cast<const Coordination::SetResponse *>(responses[SET_PROCESSING_ID].get());
-        processing_id_version = set_response->stat.version;
-        return std::pair{true, FileStatus::State::None};
-    }
+        bool create_if_not_exists_enabled = zk_client->isFeatureEnabled(DB::KeeperFeatureFlag::CREATE_IF_NOT_EXISTS);
+        if (create_if_not_exists_enabled)
+        {
+            requests.push_back(
+                zkutil::makeCreateRequest(processing_node_id_path, "", zkutil::CreateMode::Persistent, /* ignore_if_exists */ true));
 
-    if (is_request_failed(PROCESSED_PATH_DOESNT_EXIST))
-        return {false, FileStatus::State::Processed};
+            requests.push_back(zkutil::makeSetRequest(processing_node_id_path, processor_info, -1));
```

Comment:
same as above - e.g. line 69 is identical so it can be moved after if-else block